MiniBootcamp/MiniBootcamp/FeedDataManager.swift

```diff
@@ -8,11 +8,97 @@
 import Foundation
 
 protocol FeedDataManagerProtocol {
-    func fetch(completion: (Result<[TweetCellViewModel], Error>) -> Void)
+    func fetch(completion: @escaping (Result<[TweetCellViewModel], Error>) -> Void)
+}
+
+enum TweetAPIError: Error {
+    case noData
+    case response
+    case parsingData
 }
 
 class FeedDataManager: FeedDataManagerProtocol {
-    func fetch(completion: (Result<[TweetCellViewModel], Error>) -> Void) {
-        completion(.failure(NSError(domain: "", code: 0)))
+  let session: URLSession
+
+  init(session: URLSession = URLSession.shared) {
+    self.session = session
+  }
+
+    func fetch(completion: @escaping (Result<[TweetCellViewModel], Error>) -> Void) {
+      let url = URL(string: "https://gist.githubusercontent.com/ferdelarosa-wz/0c73ab5311c845fb7dfac4b62ab6c652/raw/6a39cffe68d87f1613f222372c62bd4e89ad06fa/tweets.json")
+      guard let url else { return }
+
+      let request = URLRequest(url: url)
+      let task = session.dataTask(with: request) { (data, response, error) in
+        if let error = error {
+            completion(.failure(error))
+            return
+        }
+
+
+        let decoder = JSONDecoder()
```

Comment:
I think is better to pass the `JSONDecoder` instance as a parameter instead of creating it inside this method. If you do that, you are applying dependency injection because the JSONDecoder is a dependency for your method.